components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -545,6 +560,26 @@ protected void unwrapResponse(CommonAuthResponseWrapper responseWrapper, String
         }
     }
 
+    private boolean isPromptRequest(HttpServletRequest request) {
+
+        return StringUtils.isNotBlank(request.getParameter(PROMPT_ID_PARAM))
+                && Boolean.parseBoolean(request.getParameter(PROMPT_RESP_PARAM));
+    }
+
+    /**
+     * Move to the previous node in the authentication graph.
+     *
+     * @param currentNode The current node in the authentication graph.
+     * @return The previous node in the authentication graph.
+     */
+    private AuthGraphNode moveToPreviousNode(AuthGraphNode currentNode) {
+
+        if (currentNode == null || currentNode instanceof ShowPromptNode) {
+            return currentNode;
+        }
+        return moveToPreviousNode(currentNode.getParent());
```

Comment:
Why is this done recursively? Can you explain the logic here.

Reply:
In the showPromt request, if the current node is not ShowPromptNode, then this should be a request coming from broweser's back button. Then we should go up the authentication graph until the ShowPromptNode is found